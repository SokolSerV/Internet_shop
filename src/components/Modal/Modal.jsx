import PropTypes from "prop-types";

import { ReactComponent as CrossIcon } from "./icons/cross.svg";

import "./Modal.scss";

export default function Modal({
    header,
    closeButton,
    closeModal,
    text,
    actions,
}) {
    return (
        <div
            className="modal-wrapper"
            onClick={(event) => {
                if (event.target.classList.contains("modal-wrapper")) {
                    closeModal();
                }
            }}
        >
            <div className="modal">
                <div className="modal-box">
                    {closeButton && (
                        <button
                            type="button"
                            className="modal-close"
                            onClick={() => closeModal()}
                        >
                            <CrossIcon />
                        </button>
                    )}
                    <div className="modal-header">
                        <p>{header}</p>
                    </div>
                    <div className="modal-content">
                        <p>{text}</p>
                    </div>
                    <div className="modal-footer">
                        <div className="button-wrapper">{actions}</div>
                    </div>
                </div>
            </div>
        </div>
    );
}

Modal.propTypes = {
    header: PropTypes.string,
    closeButton: PropTypes.bool,
    closeModal: PropTypes.func,
    text: PropTypes.string,
    actions: PropTypes.node,
};

Modal.defaultProps = {
    header: "Текст заголовка модального вікна",
    closeButton: false,
    closeModal: () => {},
    text: "Основний текст модального вікна",
    actions: "Код у форматі JSX",
};
