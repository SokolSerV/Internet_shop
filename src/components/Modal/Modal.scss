.modal-wrapper {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 100;
    background-color: rgba(0, 0, 0, 0.7);
}

.modal {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    max-width: 400px;
    width: 100%;
    padding: 0px;
    box-sizing: border-box;
    background-color: white;
    border-radius: 5px;

    .modal-close {
        position: absolute;
        right: 20px;
        top: 15px;
        display: inline-flex;
        vertical-align: middle;
        text-align: center;
        box-sizing: border-box;
        align-items: center;
        justify-content: center;
        outline: none;
        border: none;
        user-select: none;
        background-color: transparent;
        cursor: pointer;
    }

    .modal-header {
        padding: 15px 35px;
        background-color: lightgray;
        border-top-left-radius: 5px;
        border-top-right-radius: 5px;

        p {
            margin: 0;
            color: black;
            font-size: 20px;
            line-height: 20px;
            font-weight: 700;
        }
    }

    .modal-content {
        text-align: center;
        color: black;
        padding: 15px;
        font-size: 20px;
        line-height: 20px;

        p {
            margin: 0;
        }
    }

    .modal-footer {
        padding: 15px;
    }

    .button-wrapper {
        display: flex;
        justify-content: center;
        gap: 50px;
        align-items: center;
    }
}
