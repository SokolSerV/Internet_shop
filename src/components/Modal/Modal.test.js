import { render, screen, fireEvent } from "@testing-library/react";

import Modal from "./Modal";

describe("Modal component Smoke test (Snapshot test)", () => {
    test("should Modal render", () => {
        const { asFragment } = render(
            <Modal
                header="Test-header"
                closeButton
                text="Test-text"
                actions={<button>Click</button>}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    describe("closeButton ckick", () => {
        const closeModalMock = jest.fn();

        test("should closeModal triggering by closeButton click", () => {
            render(
                <Modal
                    closeModal={closeModalMock}
                    header="Test-header"
                    closeButton
                    text="Test-text"
                />
            );
            // eslint-disable-next-line testing-library/no-debugging-utils
            screen.debug();

            const btnCrossIcon = screen.getByRole("button");
            fireEvent.click(btnCrossIcon);

            expect(closeModalMock).toHaveBeenCalled();
        });
    });
});
