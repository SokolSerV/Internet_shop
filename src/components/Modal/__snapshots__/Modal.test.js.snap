// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Smoke test (Snapshot test) should Modal render 1`] = `
<DocumentFragment>
  <div
    class="modal-wrapper"
  >
    <div
      class="modal"
    >
      <div
        class="modal-box"
      >
        <button
          class="modal-close"
          type="button"
        >
          <svg>
            cross.svg
          </svg>
        </button>
        <div
          class="modal-header"
        >
          <p>
            Test-header
          </p>
        </div>
        <div
          class="modal-content"
        >
          <p>
            Test-text
          </p>
        </div>
        <div
          class="modal-footer"
        >
          <div
            class="button-wrapper"
          >
            <button>
              Click
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
